apps/server/src/modules/legacy-school/uc/school-system-options.uc.ts

```diff
@@ -0,0 +1,86 @@
+import { ObjectId } from '@mikro-orm/mongodb';
+import { AuthorizationContextBuilder, AuthorizationService } from '@modules/authorization';
+import { System, SystemService } from '@modules/system';
+import { Injectable } from '@nestjs/common';
+import { NotFoundLoggableException } from '@shared/common/loggable-exception';
+import { Permission } from '@shared/domain/interface';
+import { EntityId } from '@shared/domain/types';
+import { AnyProvisioningOptions, SchoolSystemOptions, SchoolSystemOptionsBuilder } from '../domain';
+import { ProvisioningOptionsInterface } from '../interface';
+import { MissingProvisioningStrategyLoggableException } from '../loggable';
+import { SchoolSystemOptionsService } from '../service';
+
+@Injectable()
+export class SchoolSystemOptionsUc {
+	constructor(
+		private readonly authorizationService: AuthorizationService,
+		private readonly systemService: SystemService,
+		private readonly schoolSystemOptionsService: SchoolSystemOptionsService
+	) {}
+
+	public async getProvisioningOptions(
+		userId: EntityId,
+		schoolId: EntityId,
+		systemId: EntityId
+	): Promise<AnyProvisioningOptions> {
+		const schoolSystemOptions: SchoolSystemOptions | null =
+			await this.schoolSystemOptionsService.findBySchoolIdAndSystemId(schoolId, systemId);
+
+		if (!schoolSystemOptions) {
+			throw new NotFoundLoggableException(SchoolSystemOptions.name, { schoolId, systemId });
+		}
+
+		const user = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkPermission(
+			user,
+			schoolSystemOptions,
+			AuthorizationContextBuilder.read([Permission.SCHOOL_SYSTEM_VIEW])
+		);
+
+		return schoolSystemOptions.provisioningOptions;
+	}
+
+	public async setProvisioningOptions(
```

Comment:
Shouldnt the name be more usecase driven then a technical set? What we are doing here is some validation, create or update and permission checks. A "set" usually suggests only changing a property

Reply:
So validateCreateOrUpdateAndCheckPermissionForProvisioningOptions? I am lacking creativity here